src/features/task-management/api/task-validation-service.ts

```
@@ -0,0 +1,36 @@
+      return err(ErrorFactory.validation("Cannot change status from DONE to other statuses"));
+    }
+    return ok(undefined);
+  }
+
+  static validatePriorityUpdate(currentStatus: any): Result<void, AppError> {
+    if (!TaskDomain.canUpdatePriority(currentStatus)) {
+      return err(ErrorFactory.validation("Cannot change priority of completed tasks"));
+    }
+    return ok(undefined);
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)